index.js

```diff
@@ -22,26 +22,43 @@ class Ora {
 			stream: process.stderr
 		}, options);
 
-		const sp = this.options.spinner;
-		this.spinner = typeof sp === 'object' ? sp : (process.platform === 'win32' ? cliSpinners.line : (cliSpinners[sp] || cliSpinners.dots)); // eslint-disable-line no-nested-ternary
-
-		if (this.spinner.frames === undefined) {
-			throw new Error('Spinner must define `frames`');
-		}
+		this.spinner = this.options.spinner;
 
 		this.color = this.options.color;
 		this.hideCursor = this.options.hideCursor !== false;
 		this.interval = this.options.interval || this.spinner.interval || 100;
 		this.stream = this.options.stream;
 		this.id = null;
-		this.frameIndex = 0;
 		this.isEnabled = typeof this.options.isEnabled === 'boolean' ? this.options.isEnabled : ((this.stream && this.stream.isTTY) && !process.env.CI);
 
 		// Set *after* `this.stream`
 		this.text = this.options.text;
 		this.linesToClear = 0;
 	}
 
+	get spinner() {
+		return this._spinner;
+	}
+
+	set spinner(spinner = cliSpinners.dots) {
```

Comment:
You should not use a default parameter here. It screw up the below conditionals.